src/test/java/teammates/test/cases/BaseTestCase.java

```diff
@@ -75,6 +75,10 @@ protected void signalFailureToDetectException(String... messages) {
         throw new RuntimeException("Expected exception not detected." + Arrays.toString(messages));
     }
 
+    protected void signalFailureToDetectAssertionError(String... messages) {
+        throw new AssertionError("Expected exception not detected." + Arrays.toString(messages));
```

Comment:
Should be `RuntimeException` rather than `RuntimeException` (and therefore, there is no need to add this method)

If you still throw `AssertionError`, the catch block will catch it (and think it is the `AssertionError` thrown by the method under test) and the test won't fail.

Reply:
>Should be RuntimeException rather than RuntimeException (and therefore, there is no need to add this method)

I don't get this?